git_theta/api.py

```diff
@@ -0,0 +1,74 @@
+"""User facing functions to interact with git-theta without incurring extra io costs."""
+
+import io
+import sys
+from typing import Optional, Union
+
+import git
+
+import git_theta
+from git_theta import checkpoints, filters, git_utils, metadata, utils
+
+
+def save(
+    state_dict,
+    path: str,
+    commit_msg: str,
+    checkpoint_type: str = "pytorch",
+    materialize: bool = False,
+) -> git.Commit:
+    """Save a model using git-theta without writing it to disk."""
```

Comment:
Should we make these into proper docstrings if we want people to be calling them from Python?

Reply:
Updated